internal/executor/util/kubernetes_object.go

```diff
@@ -89,6 +94,20 @@ func CreateIngress(name string, job *api.Job, pod *v1.Pod, service *v1.Service,
 		rules = append(rules, path)
 	}
 
+	tls := make([]networking.IngressTLS, 0, 1)
+
+	if jobConfig.TlsEnabled {
```

Comment:
what should do the validation here? E.g. is it valid to have `jobConfig.TlsEnabled = true` but `len(service.Spec.Ports = 0`.

Reply:
Are you worried about the case of people configuring no rules and just using a default backend?

Reply:
I guess what I'm really thinking about is what validates that the job spec makes sense, so that we don't go to the trouble of submitting jobs to k8s only to find out that they fail or act in a way that the user doesn't expect.

In this particular case I was wondering about the case of a user submitting a job spec with zero ports but some ingress config.  To me, this wouldn't make any sense because if you haven't got any service ports it's not clear what you'd be ingressing to (although my knowledge of k8s isn't brilliant so it's possible this is in fact a meaningful config!).

Reply:
if you have no rules everything gets routed to the default backend, which might have a use in general but probably won't be spun up (but I don't see a reason as to why we should prevent this considering it is in fact valid K8s).

Nonetheless, having a TLS certificate or not is unrelated to how many ports you configure routing rules for making validation of this completely out of the scope of this story. I hope this resolves this :)

Reply:
I guess here would be a sensible place to add the check:
https://github.com/G-Research/armada/blob/master/internal/common/validation/job.go#L13

But yeah it can be a separate PR. 

Similarly if we wanted to be pessimistic, we could also check it on the executor somewhere and skip creation on 0 ports 
 - It looks like we do in the calling function
 
However that first check would reject at submission time so the user can see instantly its invalid